### PR TITLE
Add poetry support

### PR DESCRIPTION
**Pull request by dbradf**
Move to poetry and run `black`.

### CODE REVIEW

File: evergreen.yml [hunk 1]
```diff
@@ -15,10 +15,28 @@ functions:
...
+        set -o verbose
+
+        /opt/mongodbtoolchain/v3/bin/python3 -m venv venv
         . venv/bin/activate
 
-        pip install -r requirements.txt
+        pip install poetry
+        poetry install
+
+  deploy:
```
Comment (lydiastepanek) [inline evergreen.yml]: How were we deploying before, out of curiosity?

File: evergreen.yml [hunk 2]
```diff
@@ -15,10 +15,28 @@ functions:
...
+    - command: shell.exec
+      params:
+        working_dir: src
+        script: |
+          set -o errexit
+          . venv/bin/activate
```
Comment (lydiastepanek) [inline evergreen.yml]: Do we have to activate venv if this is done in the `create virtualenv` step done as part of `pre`?

File: poetry.lock [hunk 3]
```diff
@@ -0,0 +1,475 @@
...
+
+[package.extras]
+docs = ["sphinx", "jaraco.packaging (>=3.2)", "rst.linker (>=1.9)"]
+testing = ["jaraco.itertools", "func-timeout"]
+
+[metadata]
+content-hash = "c8e106cccd62201d66bb1c81022c1baac498fc89adb4b1ce939e087b000e4a9f"
+python-versions = "^3.7"
+
+[metadata.files]
```
Comment (lydiastepanek) [inline poetry.lock]: this is cool

File: evergreen.yml (hunk 1 above)
Comment (dbradf) [reply evergreen.yml]: manually running the commands.

File: evergreen.yml (hunk 2 above)
Comment (dbradf) [reply evergreen.yml]: I think so. I believe each command is run in its own shell and `activate` only applies to the current shell.

**Comment by dbradf** [comment]
evergreen merge

**Pull request merged**